pkg/krmtotf/tftokrm.go

```diff
@@ -80,26 +80,51 @@ func GetSpecAndStatusFromState(resource *Resource, state *terraform.InstanceStat
 		}
 		target := &spec
 		if !fieldSchema.Required && !fieldSchema.Optional {
-			target = &status
-			key = renameStatusFieldIfNeeded(resource.ResourceConfig.Name, key)
+			if k8s.OutputOnlyFieldsAreUnderObservedState(resource.Kind, resource.GroupVersionKind().Version) {
+				observedState, ok := status[k8s.ObservedStateFieldName]
+				if !ok {
+					// Always add the 'observedState' subfield if the resource
+					// should have its computed field under the observed state.
+					observedState = make(map[string]interface{})
+					status[k8s.ObservedStateFieldName] = observedState
+				}
+				observedStateMap := observedState.(map[string]interface{})
+				target = &(observedStateMap)
+			} else {
+				target = &status
+				key = renameStatusFieldIfNeeded(resource.ResourceConfig.Name, key)
+			}
 		}
 		(*target)[key] = val
 	}
 	if location, ok := getLocationValueFromResourceOrState(resource, unmodifiedState); ok {
 		spec["location"] = location
 	}
-	if conditions, ok := resource.Status["conditions"]; ok {
+	if conditions, ok := resource.GetStatus()["conditions"]; ok {
 		status["conditions"] = deepcopy.DeepCopy(conditions)
 	}
-	if observedGeneration, ok := resource.Status["observedGeneration"]; ok {
+	if observedGeneration, ok := resource.GetStatus()["observedGeneration"]; ok {
 		status["observedGeneration"] = deepcopy.DeepCopy(observedGeneration)
 	}
 	if resource.ResourceConfig.ObservedFields != nil {
-		observedState := resolveObservedState(resource, krmState)
-		if len(observedState) > 0 {
-			status["observedState"] = observedState
+		observedFields := resolveObservedFields(resource, krmState)
+		if len(observedFields) > 0 {
+			// Merge the observed fields into the observed state.
```

Comment:
Should we merge or just build it anew?

Reply:
Just to clarify, what is "build it anew" in this context? Creating a new map?

Observed fields are also part of the observed state but manually configured for `spec` fields in service mappings (#1023, #1059) so I think merging should be the right behavior. Glad to adjust if you are suggesting a simpler implementation.